### PR TITLE
TEAM/HIER: 2lvl alltoall

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**

### CODE REVIEW

File: src/team_lib/hier/xccl_hier_lib.c [hunk 1]
```diff
@@ -101,6 +101,48 @@ static xccl_status_t xccl_hier_open(xccl_team_lib_h self,
...
+            spec.pairs.node = XCCL_HIER_PAIR_NODE_UCX;
+            spec.pairs.flat = XCCL_HIER_PAIR_FLAT_UCX;
+            break;
+        case UCS_MEMORY_TYPE_CUDA:
+            if (ctx->tls[ucs_ilog2(XCCL_TL_NCCL)].enabled) {
+                spec.pairs.node = XCCL_HIER_PAIR_NODE_NCCL;
+                spec.pairs.flat = XCCL_HIER_PAIR_FLAT_UCX;
+            } else {
+                spec.pairs.node = XCCL_HIER_PAIR_NODE_UCX;
+                spec.pairs.flat = XCCL_HIER_PAIR_FLAT_UCX;
```
Comment (vspetrov) [inline src/team_lib/hier/xccl_hier_lib.c]: why not to move spec.pairs.flat = XCCL_HIER_PAIR_FLAT_UCX outside of all conditionals (before switch). Looks like it is always UCX there.

File: src/team_lib/hier/xccl_hier_schedule.c [hunk 2]
```diff
@@ -333,6 +334,10 @@ xccl_status_t hier_task_progress_handler(xccl_coll_task_t *task)
     assert(self->req);
     for (i = 0; (i < n_polls) && (task->state == XCCL_TASK_STATE_INPROGRESS); i++) {
         status = xccl_collective_test(self->req);
+        if (self->scratch != NULL) {
```
Comment (vspetrov) [inline src/team_lib/hier/xccl_hier_schedule.c]: should this be in the completed branch?

**Review by vspetrov — APPROVED** [review]

File: src/team_lib/hier/xccl_hier_lib.c (hunk 1 above)
Comment (Sergei-Lebedev) [reply src/team_lib/hier/xccl_hier_lib.c]: fixed

File: src/team_lib/hier/xccl_hier_schedule.c (hunk 2 above)
Comment (Sergei-Lebedev) [reply src/team_lib/hier/xccl_hier_schedule.c]: yes! thanks
moved to wrong place during rebase

**Pull request merged**